src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.cpp

```diff
@@ -130,211 +52,49 @@ std::string BrgemmKernelConfig::to_string() const {
-        auto check_port = [&](const ov::snippets::lowered::LoopPort& p) { return p.dim_idx == 0; };
-        OPENVINO_ASSERT(in_ports.size() >= 2 && !in_ports.front().is_incremented && std::all_of(in_ports.cbegin(), in_ports.cend(), check_port) &&
-                        out_ports.size() == 1 && check_port(out_ports.back()),
-                        "Incorrect Loop by Brgemm dimension N");
-        N = current_expanded_loop_info->get_increment();
-        input_pds[1]->set_subtensor_dim(0, N);
-        output_pds[0]->set_subtensor_dim(0, N);
-    }
-
-    /* ------- Dimension K ----------*/
-    // 1. If Brgemm block processes full dimension K -> `beta = 0`
-    // 2. If Brgemm block processes part of the dimension K (there is blocking), need to find
-    //    the most first executed Brgemm Block in Loops which iterate through dimension K (work_amount > 0).
-    //    First of them will have `beta = 0`, other - `beta = 1`
-    float beta = 0;
-    if (ov::snippets::utils::is_full_dim_value(K)) {
-        K = *in0_shape.rbegin();
-    } else {
-        const auto& current_expanded_loop_info = get_loop_info();
-        const auto& in_ports = current_expanded_loop_info->get_input_ports();
-        const auto& out_ports = current_expanded_loop_info->get_output_ports();
-        // Quick validation check: Should we check that port is really Brgemm port?
-        // Note: We check `is_incremented` attribute only for not incremented ports because
-        //       this `is_incremented = true` can be changed by `CleanRepeatedDataPointerShifts` optimization
-        OPENVINO_ASSERT(in_ports.size() >= 2 && in_ports.front().dim_idx == 0 && in_ports.back().dim_idx == 1 &&
-                        out_ports.size() == 1 && !out_ports.front().is_incremented,
-                        "Incorrect Loop by Brgemm dimension K");
-        K = current_expanded_loop_info->get_increment();
-        input_pds[0]->set_subtensor_dim(0, K);
-        input_pds[1]->set_subtensor_dim(1, K);
-        if (K > 0)
-            beta = get_beta(loop_manager, static_cast<int>(loop_ids.back()), current_expanded_loop_info);
-    }
-
-    const auto LDA = DIM_CAST(snippets::utils::get_dim_stride(expr->get_input_port(0)));
-    const auto LDC = DIM_CAST(snippets::utils::get_dim_stride(expr->get_output_port(0)));
-    auto LDB = DIM_CAST(snippets::utils::get_dim_stride(expr->get_input_port(1)));
-    const auto& brgemm_node = as_type_ptr<ov::intel_cpu::BrgemmCPU>(expr->get_node());
-    OV_CPU_JIT_EMITTER_ASSERT(brgemm_node, "Got invalid node type in update_config");
-    // In case of data repacking LDB is chosen in accordance with repacking buffer size
-    if (with_repacking(brgemm_node->get_type()))
-        LDB = brgemm_utils::repacking::compute_LDB(LDB, brgemm_node->get_input_element_type(1));
-
-    config.update(DIM_CAST(M), DIM_CAST(N), DIM_CAST(K), LDA, LDB, LDC, beta);
+    return BrgemmBaseKernelExecutor::update_config(expr, linear_ir, config);
```

Comment:
If I got it right, unfortunately, we cannot do so because I use multiple inheritance here:
```
BrgemmBaseKernelExecutor           CPUKernelExecutor<Conf, KernelType>
                        \                                   /
      BrgemmKernelExecutor or BrgemmAMXKernelExecutor 
```

It's needed to store only [one pointer](https://github.com/a-sidorova/openvino/blob/bb4ca7bc0390725915449c5c8ee8ed0bae62a604/src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_emitter.hpp#L31) to executor in `jit_brgemm_emitter`: `std::shared_ptr<BrgemmBaseKernelExecutor> m_executor` (we cannot store pointer of template class without params).

Then `CPUKernelExecutor<Conf, KernelType>` has abstract method `update_config()`, this impl is the same for `BrgemmKernelExecutor` and `BrgemmAMXKernelExecutor`. So I create this common method `BrgemmBaseKernelExecutor ::update_config()` which is called in methods of each class

Correct me please if I missed something!

Thank you!

Reply:
Git it, thanks for the details. Can we inherit `BrgemmBaseKernelExecutor<Conf, KernelType>` from `CPUKernelExecutor<Conf, KernelType>` maybe?